srv/GetEntities.srv

```diff
@@ -0,0 +1,12 @@
+# Get objects in the scene which can be interacted, e.g. with using SetEntityState.
+
+string filter                                       # Optional, defaults to empty. Return entities with matching names.
+                                                    # Entity names are matched with the filter regular expression.
+                                                    # An empty filter will result in all entities being returned.
```

Comment:
Looking at this, another cool idea could be to tag `Entity`s with some category label, e.g., "robot" vs. "object" vs. "furniture", etc.

This way you can not only filter by name, but also by category -- of course, every simulator interface implementer would tag their own entities with relevant categories.

Reply:
I like this idea but it seems to be introducing a new concept of entity tags. Which needs to be reflected in the Entity.msg and other interfaces. Perhaps this could be an extension? This sounds much like a synthetic data feature.

Reply:
@sea-bass what do you think? Would you like to include labels in the PR still?

Reply:
I think it would be useful to label entities with categories (as you said in the `Entity.msg`), and then be able to retrieve them by categories.

Should it be done with this initial PR? Not necessarily, but I'd appreciate at least a separate gitissue tracking this request. I certainly use these mechanisms all the time in my simulator.

Reply:
I added categories and tags to this PR. They are in EntityInfo now, and can be queried with GetEntityInfo, and also filtering now extends to tags (any or all) and categories (any of).